tests/library/test_readme.py

```diff
@@ -0,0 +1,166 @@
+import re
+import unittest
+from collections import defaultdict
+from typing import Dict, List, Optional, Tuple
+
+from recipe_scrapers import SCRAPERS, AbstractScraper
+
+START_LIST = "-----------------------"
+END_LIST = "(*) offline saved files only"
+
+ScraperIndex = Dict[str, Tuple[AbstractScraper, List[str]]]
+
+
+def get_scraper_domains():
+    scraper_domains = defaultdict(list)
+    for domain, scraper in SCRAPERS.items():
+        primary_domain = scraper.host()
+        if domain == primary_domain:
+            scraper_domains[scraper].insert(0, domain)
+        else:
+            scraper_domains[scraper].append(domain)
+    return scraper_domains
+
+
+def get_scraper_index() -> ScraperIndex:
+    scraper_index = {}
+    for scraper, domains in get_scraper_domains().items():
+        shared_prefix = get_shared_prefix(domains)
+
+        if not shared_prefix:
+            # Treat all as primary domains
+            for domain in domains:
+                scraper_index[domain] = (scraper, [domain])
+            continue
+
+        # Index the primary domain and include their secondary domains minus the shared prefix
+        primary_domain = scraper.host()
+        scraper_index[primary_domain] = (
+            scraper,
+            [
+                domain.removeprefix(shared_prefix)
+                for domain in domains
+                if domain != shared_prefix
+            ],
+        )
```

Comment:
The removeprefix method was introduced in Python 3.9, so our earlier version tests won't pass. This change rec is an equivalent implementation that _should_ pass all tests

```suggestion
domain[len(shared_prefix):] if domain.startswith(shared_prefix) else domain
```

Reply:
Yep, makes sense.  Let's also consider dropping support for py3.8 in the not-too-distant future, since it's reaching end-of-support in October this year according to the [Python downloads page](https://www.python.org/downloads/).